inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -98,12 +98,87 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
     record_ids = [el['_source']['control_number'] for el in matches]
-    if record_ids:
-        obj.extra_data['record_matches'] = record_ids
-        return True
+    obj.extra_data.setdefault('matches', {})['exact'] = record_ids
+    return bool(record_ids)
 
-    obj.extra_data['record_matches'] = []
-    return False
+
+@with_debug_logging
+def fuzzy_match(obj, eng):
+    """Return ``True`` if a similar record is found in the system.
+
+    Uses a custom configuration for ``inspire-matcher`` to find records
+    similar to the current workflow object's payload in the system.
+
+    Also sets the ``matches.fuzzy`` property in ``extra_data`` to the list of
+    control numbers that matched.
+
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the workflow object has a duplicate in the system
+        ``False`` otherwise.
+
+    """
+    fuzzy_match_configuration = {
+        'algorithm': [
+            {
+                'queries': [
+                    {
+                        'clauses': [
+                            {
+                                'boost': 20,
+                                'path':  'abstracts',
+                            },
+                            {
+                                'boost': 10,
+                                'path':  'authors[:3]',
+                            },
+                            {
+                                'boost': 20,
+                                'path':  'titles',
+                            },
+                            {
+                                'boost': 10,
+                                'path':  'report_numbers',
+                            },
+                        ],
+                        'type': 'fuzzy',
+                    }
+                ]
+            }
+        ],
+        'doc_type': 'hep',
+        'index': 'records-hep',
+    }
+
+    matches = dedupe_list(match(obj.data, fuzzy_match_configuration))
+    record_ids = [el['_source']['control_number'] for el in matches]
+    obj.extra_data.setdefault('matches', {})['fuzzy'] = record_ids
+    return bool(record_ids)
+
+
+@with_debug_logging
+def is_fuzzy_match_approved(obj, eng):
+    """Check if a fuzzy match has been approved.
+
+    Also, if the property `fuzzy_match_approved_id` in obj.extra_data
+    contains an id, it is added to the property `matches.approved` in
+    obj.extra_data.
+
+
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the property fuzzy_match_approved_id
+        in obj.extra_data contains an integer id, ``False`` otherwise.
+    """
+    approved_match_id = obj.extra_data.get('fuzzy_match_approved_id')
+    obj.extra_data.setdefault('matches', {})['approved'] = approved_match_id
+    return bool(isinstance(approved_match_id, (int, long)))
```

Comment:
the bool is not needed here, isinstance always returns bool.

Reply:
The fact that you are checking the type of the value looks strange. Should we care about that? If it's a string or anything? I think we should only care if there's any, not what it is.

Reply:
Depends on how we define the API. If we specify the front-end has to pass an integer, then we can check if it is compliant to the spec.